web/auth/auth.go

```diff
@@ -541,21 +541,37 @@ func authorizeForm(c echo.Context) error {
 			"Error":  "Error Invalid scope",
 		})
 	}
+  readOnly := true
+  for _, p := range permissions {
+    if !p.Verbs.ReadOnly() {
+      readOnly = false
+    }
+  }
```

Comment:
Oh, indentation looks weird. Can you use [gofmt](https://blog.golang.org/go-fmt-your-code) on this file?